Tests/TestKeppPrmissions.m

```diff
@@ -0,0 +1,69 @@
+//
+//  TestKeppPrmissions.m
+//  UnzipKitTests
+//
+//  Created by MartinLau on 2019/6/13.
+//  Copyright © 2019 Abbey Code. All rights reserved.
+//
+
+#import "UZKArchiveTestCase.h"
+#import "zip.h"
+#import "UnzipKit.h"
+
+
+@interface TestKeppPrmissions : UZKArchiveTestCase
+
+@end
+
+@implementation TestKeppPrmissions
+
+- (void)setUp {
+    // Put setup code here. This method is called before the invocation of each test method in the class.
+    [super setUp];
+}
+
+- (void)tearDown {
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+}
+
+- (void)testExtract {
+    // This is an example of a functional test case.
+    // Use XCTAssert and related functions to verify your tests produce the correct results.
+    
+    UZKArchive *archive = [[UZKArchive alloc] initWithURL:self.testFileURLs[@"Test Permissions Archive.zip"] error:nil];
+    NSString *extractDirectory = [self randomDirectoryWithPrefix:archive.filename.stringByDeletingPathExtension];
+    
+    NSError *extractError = nil;
+    BOOL success = [archive extractFilesTo:extractDirectory overwrite:NO error:&extractError];
+    XCTAssert(success, @"extract error %@", extractError);
+    // $ ls -l "extractURL.path" look up prmissions
+}
+
+- (void)testWriteFile {
+    
```

Comment:
Since we're getting rid of `writeFile`, please replace this with a couple other tests that call `writeData`: One that specifies permissions, and one that doesn't. Then you can extract the archived file two ways in each test to verify the permissions: use `UZKArchive`, and also extract using the `zip` shell command, then verify the permissions of the extracted files using `NSFileManager`.

`-[UZKArchiveTestCase extractArchive:password]` is already written, and will perform the shell extraction. I can help if you have any trouble.

Reply:
Next, I may be unable to participate in the revision due to work problems. I need your help :(

Reply:
I can make a branch for this PR off of `v1.9` and put these test cases into that branch, if you'd like. You can then target that branch instead of `v1.9` for the PR, or at least rebase your changes off of that branch, so my changes become part of the PR. Your choice (either way is fine with me).

Reply:
I agree with you, this will help me to continue to change.